xray/utils/resultwriter.go

```diff
@@ -52,40 +62,130 @@ func PrintScanResults(results []services.ScanResponse, errors []formats.SimpleJs
+			return err
 		}
-		jsonTable.Errors = errors
 		return printJson(jsonTable)
 	case Json:
 		return printJson(results)
+	case Sarif:
+		sarifFile, err := GenerateSarifFileFromScan(results, includeVulnerabilities, isMultipleRoots)
+		if err != nil {
+			return err
+		}
+		log.Output(sarifFile)
+	}
+	return nil
+}
+
+func GenerateSarifFileFromScan(currentScan []services.ScanResponse, includeVulnerabilities, isMultipleRoots bool) (string, error) {
+	report, err := sarif.New(sarif.Version210)
+	if err != nil {
+		return "", errorutils.CheckError(err)
+	}
+	run := sarif.NewRunWithInformationURI("Xray", "https://jfrog.com/xray/")
+	err = convertScanToSarif(run, currentScan, includeVulnerabilities, isMultipleRoots)
+	if err != nil {
+		return "", err
+	}
+	report.AddRun(run)
+	out, err := json.Marshal(report)
+	if err != nil {
+		return "", errorutils.CheckError(err)
+	}
+
+	return clientUtils.IndentJson(out), nil
+}
+
+func convertScanToSimpleJson(results []services.ScanResponse, errors []formats.SimpleJsonError, includeVulnerabilities, isMultipleRoots, includeLicenses bool) (formats.SimpleJsonResults, error) {
+	violations, vulnerabilities, licenses := splitScanResults(results)
+	jsonTable := formats.SimpleJsonResults{}
+	if includeVulnerabilities {
+		log.Info(noContextMessage + "All vulnerabilities detected will be included in the output JSON.")
+		vulJsonTable, err := PrepareVulnerabilities(vulnerabilities, isMultipleRoots)
+		if err != nil {
+			return formats.SimpleJsonResults{}, err
+		}
+		jsonTable.Vulnerabilities = vulJsonTable
+	} else {
+		secViolationsJsonTable, licViolationsJsonTable, opRiskViolationsJsonTable, err := PrepareViolations(violations, isMultipleRoots)
+		if err != nil {
+			return formats.SimpleJsonResults{}, err
+		}
+		jsonTable.SecurityViolations = secViolationsJsonTable
+		jsonTable.LicensesViolations = licViolationsJsonTable
+		jsonTable.OperationalRiskViolations = opRiskViolationsJsonTable
+	}
+
+	if includeLicenses {
+		licJsonTable, err := PrepareLicenses(licenses, isMultipleRoots)
+		if err != nil {
+			return formats.SimpleJsonResults{}, err
+		}
+		jsonTable.Licenses = licJsonTable
+	}
+	jsonTable.Errors = errors
+
+	return jsonTable, nil
+}
+
+func convertScanToSarif(run *sarif.Run, currentScan []services.ScanResponse, includeVulnerabilities, isMultipleRoots bool) error {
+	var errors []formats.SimpleJsonError
+	jsonTable, err := convertScanToSimpleJson(currentScan, errors, includeVulnerabilities, isMultipleRoots, false)
+	if err != nil {
+		return err
 	}
+	if len(jsonTable.SecurityViolations) > 0 {
+		violations := jsonTable.SecurityViolations
+		licenses := jsonTable.LicensesViolations
+		for i := 0; i < len(jsonTable.SecurityViolations); i++ {
+			impactedPackageFull := violations[i].ImpactedPackageName + " " + violations[i].ImpactedPackageVersion
+			if violations[i].FixedVersions != nil {
+				violations[i].Summary += "\n . Fixed in Versions: " + strings.Join(violations[i].FixedVersions, ",")
```

Comment:
Is "Fixed in Versions" in the title? If so, can we move it somewhere else?

Reply:
Not in the title, but in the Summary (which is displayed when the user gets into the relevant security issue)